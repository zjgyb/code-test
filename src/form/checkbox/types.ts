/*
 * @Author: gyb
 * @Date: 2021-10-29
 * @Description: checkbox组件类型声明文件
 */

import type { IxPublicPropTypes } from '../../table/types';

export const checkboxProps = {
    checked: {
        type: Boolean,
        default: false
    },

    // 是否半选状态
    isHalf: {
        type: Boolean,
        default: false
    }
}

export type CheckboxPublicProps = Required<IxPublicPropTypes<typeof checkboxProps>>;