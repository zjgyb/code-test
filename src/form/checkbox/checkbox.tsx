/*
 * @Author: gyb
 * @Date: 2021-10-29
 * @Description: checkboxç»„ä»¶
 */

import type { CheckboxPublicProps } from './types';

import { computed, defineComponent } from '@vue/composition-api';
import { checkboxProps } from './types';

export default defineComponent({
	name: 'SfCheckbox',
	props: checkboxProps,
	setup(props: CheckboxPublicProps, context) {
		const onCheckboxChange = (event: Event) => {
			const value = (event.target as HTMLInputElement).checked;
			context.emit('update:checked', value);
			context.emit('change', value);
		}

		const classes = computed(() => {
			return {
				'sf-checkbox_inner': true,
				'sf-checkbox_inner--checked': props.checked,
				'sf-checkbox_inner--half': props.isHalf,
				'sf-checkbox_inner--disabled': props.disabled
			}
		});

		return () => {
			return (
				<label class="sf-checkbox">
					<span class={classes.value}></span>
					<input 
						class="sf-checkbox_input"
						type="checkbox"
						checked={props.checked}
						onChange={onCheckboxChange}
						disabled={props.disabled} />
				</label>
			)
		}
	}
});
