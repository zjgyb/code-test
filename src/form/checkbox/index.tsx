/*
 * @Author: gyb
 * @Date: 2021-10-29
 * @Description: checkbox组件模块导出
 */

import SfCheckbox from './checkbox';

export { SfCheckbox }
