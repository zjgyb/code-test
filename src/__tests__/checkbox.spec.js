import { mount } from '@vue/test-utils'
import { SfCheckbox } from '../form/checkbox'

describe('Pagination', () => {
    const CheckboxMount = options => mount(SfCheckbox, options)

    test('render', () => {
        const wrapper = CheckboxMount()
        expect(wrapper.html()).toMatchSnapshot()
        expect(() => {
            wrapper.vm.$forceUpdate()
            wrapper.vm.$destroy()
        }).not.toThrow()
    });

    test('checked', async() => {
        const wrapper = CheckboxMount({
            propsData: {
                checked: false
            }
        })
        const checkbox = wrapper.find('.sf-checkbox input');
        expect(checkbox.exists()).toBeTruthy();
        await checkbox.setChecked();
        expect(checkbox.element.checked).toBeTruthy();
    });

    test('half checked', async() => {
        const wrapper = CheckboxMount({
            propsData: {
                checked: false,
                isHalf: true
            }
        })
        const checkbox = wrapper.find('.sf-checkbox_inner--half');
        expect(checkbox.exists()).toBeTruthy();
    });
})