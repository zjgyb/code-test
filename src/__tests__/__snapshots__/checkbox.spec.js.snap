// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render 1`] = `"<label class=\\"sf-checkbox\\"><span class=\\"sf-checkbox_inner\\"></span><input type=\\"checkbox\\" class=\\"sf-checkbox_input\\"></label>"`;
